x/concentrated-liquidity/position.go

```diff
@@ -478,7 +478,19 @@ func (k Keeper) fungifyChargedPosition(ctx sdk.Context, owner sdk.AccAddress, po
 			}
 			// If the accumulator contains the position, move the unclaimed rewards to the new position.
 			if hasPosition {
-				// Prepare the accumulator for the old position.
+				// Get number of share in the position.
+				positionShares, err := uptimeAccum.GetPositionSize(oldPositionName)
+				if err != nil {
+					return 0, err
+				}
+
+				// Remove the position from the accumulator so that it is deleted when
+				// rewards are claimed.
+				if err := uptimeAccum.RemoveFromPosition(oldPositionName, positionShares); err != nil {
+					return 0, err
+				}
+
```

Comment:
good catch! Would it be possible to add tests cases that contians this change please?

Reply:
So this is already covered by test cases in general.

I have this particular bug reproduced in a test case in another branch. Unfortunately, that test case depends on the change that I haven't yet merged in.

Ref: https://github.com/osmosis-labs/osmosis/issues/5038

I will PR this repro test as soon as I address #5038